pkg/operator/ceph/object/spec_test.go

```diff
@@ -876,3 +876,50 @@ func TestAWSServerSideEncryption(t *testing.T) {
 		assert.True(t, checkRGWOptions(rgwContainer.Args, c.sseS3VaultTLSOptions(true)))
 	})
 }
+
+func TestAddDNSNamesToRGWPodSpec(t *testing.T) {
+	store := simpleStore()
+	info := clienttest.CreateTestClusterInfo(1)
+	data := cephconfig.NewStatelessDaemonDataPathMap(cephconfig.RgwType, "default", "rook-ceph", "/var/lib/rook/")
+
+	tests := []struct {
+		name           string
+		dnsNames       []string
+		expectedDNSArg string
+		cephvers       cephver.CephVersion
+		wantErr        bool
+	}{
+		{"no dns names", []string{}, "", cephver.CephVersion{Major: 18, Minor: 0, Extra: 0}, false},
+		{"one dns name ceph v18", []string{"my.dns.name"}, "--rgw-dns-name=my.dns.name", cephver.CephVersion{Major: 18, Minor: 0, Extra: 0}, false},
+		{"multiple dns names ceph v18", []string{"my.dns.name1", "my.dns.name2"}, "--rgw-dns-name=my.dns.name1,my.dns.name2", cephver.CephVersion{Major: 18, Minor: 0, Extra: 0}, false},
+		{"one dns name ceph v17", []string{"my.dns.name"}, "--rgw-dns-name=my.dns.name", cephver.CephVersion{Major: 17, Minor: 0, Extra: 0}, false},
+		{"multiple dns names ceph v17", []string{"my.dns.name1", "my.dns.name2"}, "", cephver.CephVersion{Major: 17, Minor: 0, Extra: 0}, true},
+		{"invalid dns name ", []string{"MY.DNS.COM"}, "", cephver.CephVersion{Major: 18, Minor: 0, Extra: 0}, true},
```

Comment:
```suggestion
		{"invalid dns name ceph v18", []string{"MY.DNS.COM"}, "", cephver.CephVersion{Major: 18, Minor: 0, Extra: 0}, true},
```

Add additional tests to be sure version doesn't affect none/one/multiple/invalid processing. Also add tests for Ceph greater than Reef. We've seen issues where something works for the specific major version but not newer major versions.
 - Ceph v18, mixed invalid and valid
 - Ceph version 19, none, one, multiple, one invalid, mixed invalid
 - Ceph v17, none, invalid, mixed invalid